erpnext/stock/doctype/stock_entry/stock_entry.js

```diff
@@ -15,6 +14,25 @@ frappe.ui.form.on('Stock Entry', {
 			}
 		});
 
+		frappe.db.get_value('Stock Settings', {name: 'Stock Settings'}, 'sample_retention_warehouse', (r) => {
+			if (r.sample_retention_warehouse) {
+				frm.set_query("from_warehouse", function() {
```

Comment:
There is already another setup_query for warehouse fields in `stock_controller.js`. It will overwrite that function.

https://github.com/frappe/erpnext/blob/develop/erpnext/public/js/controllers/stock_controller.js#L10

Reply:
@nabinhait Only way I can think of is to have same filters in stock_entry.js since we need them only for source warehouse. Cannot change the core filters on warehouse since it is being used in other doctypes as well. Let me know if it ok.